src/components/alert/alert.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useMemo } from 'react'
+import clsx from 'clsx'
+import {
+  InformationCircleSolid,
+  CheckCircleSolid,
+  XCircleSolid,
+  ExclamationSolid,
+  XSolid,
+} from 'components/icons'
+import Button from 'components/button'
+
+export enum AlertType {
+  info = 'info',
+  success = 'success',
+  warning = 'warning',
+  error = 'error',
+}
+
+/**
+ * Alert properties
+ */
+export type AlertProps = {
+  /** Type of the alert which can be either `AlertType.error` or `AlertType.warning` or `AlertType.success` or `AlertType.info` */
+  type?: AlertType
+  /** Alert title */
+  title: React.ReactNode
+  /** Alert content */
+  content?: React.ReactNode
+  /** Alert icon */
+  icon?: JSX.Element | boolean
+  /** Alert action buttons */
+  actions?: JSX.Element
+  /** Making alert closable. It would render (X) button */
+  closable?: boolean
+  /** Function to be called on pressing the (X) button */
+  onClose?: (event: React.MouseEvent<HTMLButtonElement>) => void
+}
+
+/** Alert component to render `info`, `warning`, `success` and `error` messages */
+export function Alert({
+  type = AlertType.info,
+  title,
+  content,
+  icon,
+  actions,
+  closable = false,
+  onClose,
+}: AlertProps) {
+  // Function to return icon, based on user input and alert type
+  const renderIcon = useMemo(() => {
+    if (icon === false) {
+      return
+    }
+
+    if (icon) {
+      return icon
+    }
+
+    if (type === AlertType.info) {
+      return <InformationCircleSolid className="w-6 h-6 text-blue-500" />
+    } else if (type === AlertType.success) {
+      return <CheckCircleSolid className="w-6 h-6 text-green-500" />
+    } else if (type === AlertType.warning) {
+      return <ExclamationSolid className="w-6 h-6 text-yellow-400" />
+    } else if (type === AlertType.error) {
+      return <XCircleSolid className="w-6 h-6 text-red-500" />
+    }
+  }, [icon, type])
+
+  // function to return required background color based on Alert Type
+  const alertBackground = useMemo(() => {
+    if (type === AlertType.info) {
+      return 'bg-blue-100'
+    } else if (type === AlertType.success) {
+      return 'bg-green-100'
+    } else if (type === AlertType.warning) {
+      return 'bg-yellow-100'
+    } else if (type === AlertType.error) {
+      return 'bg-red-100'
+    }
+  }, [type])
+
+  const textColor = useMemo(() => {
```

Comment:
we don't need `textColor` anymore as we can directly use `text-${baseColor}-700` directly.